digital_twins/incubator-monitor-server/lifecycle/execute.py

```diff
@@ -0,0 +1,225 @@
+import subprocess, os, time, sys, time
+incubator_location = os.getenv("INCUBATOR_PATH")
+sys.path.append(os.path.join(os.getcwd() + f"/{incubator_location}"))
+from threading import Thread, Event
+from omniORB import CORBA
+from omniORB.any import to_any
+import Monitor
+import CosNaming
+from incubator.communication.server.rabbitmq import Rabbitmq
+from incubator.config.config import load_config
+from digital_twin.communication.rabbitmq_protocol import ROUTING_KEY_ENERGY_SAVER_ENABLE, ROUTING_KEY_ENERGY_SAVER_STATUS, ROUTING_KEY_LIDOPEN
+
+
+def startNuRV(ior):
+    # overwrite relevant commands
+    with open(os.getcwd() + "/../commands", "r+") as f:
+        lines = f.readlines()
+        lines[-1] = f"monitor_server -N {ior}"
+        f.seek(0)
+        f.writelines(lines)
+        f.close()
+    # Start NuRV monitor server    
+    nurvPath = os.getenv("NURV_PATH")
+    nurvProcess = subprocess.Popen(f"exec {nurvPath}/NuRV_orbit -source ../commands ../safe-operation.smv", shell=True, cwd=os.getcwd(), stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
```

Comment:
The ORBit name server executable is now installed as part of the python venv (since it is included in a python package).  I was not aware of a "services" folder, and as far as I can see, it does not currently exist. As for NuRV, yes it is external to the DT and not really a part of it as they can be run independently of each other. When starting the service, the RV specification does however need to be specified as far as i can see. So I think it has more of a tool feel (as the service is started for a single specification and shut down when the DT is stopped). We can however still parameterize the startup and move it to a /common/services/  folder if you still think this is beneficial.

Reply:
Please go ahead and create **service** folder.

We are considering two scenarios for NuRV.

1. A NuRV service that is started and stopped with DT (both happen via execute and terminate scripts).
2. A NuRV service that is started outside of DT itself as an independent service. You can place `services/NuRV/start.sh` and `services/NuRV/stop.sh`. This service will then be reusable for two or more instances of DT. Whether these DTs run in sequence or concurrently is based on the ability of NuRV to handle such scenarios. At the least, a sequential servicing of DTs by NuRV service is needed before claiming the ability of NuRV to run as an independent monitoring service.

I don't know of the way NuRV takes monitoring conditions. If these can be parameterized and given at execution time, then scope of reuse is more. Otherwise, we will use it as it is.